core/src/main/java/org/elasticsearch/transport/netty/NettyTransportChannel.java

```diff
@@ -134,6 +141,14 @@ public void sendResponse(Throwable error) throws IOException {
         transportServiceAdapter.onResponseSent(requestId, action, error);
     }
 
+    private void close() {
```

Comment:
> I don't follow this one? this is about freeing the bytes of the request not the bytes of the response?

Yes, I'm sorry, I confused myself and brought something irrelevant into this discussion.

> I'm good with doing it in a finally clause, though I think close() indicates well that the channel is closed and shouldn't be used again.

That's why I don't like it, we are "closing" and then still proceeding to perform operations on the channel. I don't like the semantics. Can I suggest `release` or `unreserve`?